go-chaos/internal/deployment.go

```diff
@@ -24,28 +24,19 @@ import (
 )
 
 func (c K8Client) getGatewayDeployment() (*v12.Deployment, error) {
+
 	listOptions := metav1.ListOptions{
-		LabelSelector: getSelfManagedGatewayLabels(),
+		LabelSelector: c.getGatewayLabels(),
 	}
 	deploymentList, err := c.Clientset.AppsV1().Deployments(c.GetCurrentNamespace()).List(context.TODO(), listOptions)
 	if err != nil {
 		return nil, err
 	}
 
+	// here it is currently hard to distingush between not existing and embedded gateway;
+	// since we don't use embedded gateway in our current chaos setup I would not support it right now here
 	if deploymentList == nil || len(deploymentList.Items) <= 0 {
-		// lets check for SaaS setup
-		listOptions.LabelSelector = getSaasGatewayLabels()
-		deploymentList, err = c.Clientset.AppsV1().Deployments(c.GetCurrentNamespace()).List(context.TODO(), listOptions)
-		if err != nil {
-			return nil, err
-		}
-
-		// here it is currently hard to distingush between not existing and embedded gateway;
-		// since we don't use embedded gateway in our current chaos setup I would not support it right now here
-		if deploymentList == nil || len(deploymentList.Items) <= 0 {
-			return nil, errors.New(fmt.Sprintf("Expected to find standalone gateway deployment in namespace %s, but none found!", c.GetCurrentNamespace()))
-		}
+		return nil, errors.New(fmt.Sprintf("Expected to find standalone gateway deployment in namespace %s, but none found!", c.GetCurrentNamespace()))
```

Comment:
:wrench: As per the comment, I would add a small bit after saying we don't support embedded gateway, e.g.

```go
return nil, errors.New(fmt.Sprintf("Expected to find standalone gateway deployment in namespace %s, but none found! The embedded gateway is not supported", c.GetCurrentNamespace()))
```

Reply:
Good point thanks